src/content/learn/thinking-in-react.md

```diff
@@ -437,36 +433,37 @@ td {
 
 </Sandpack>
 
-Notice that editing the form doesn't work yet. There is a console error in the sandbox above explaining why:
+Kumbuka kwamba kuhariri fomu bado haifanyi kazi. Kuna hitilafu ya kiweko kwenye sanduku la mchanga hapo juu kuelezea kwa nini:
 
 <ConsoleBlock level="error">
 
-You provided a \`value\` prop to a form field without an \`onChange\` handler. This will render a read-only field.
+Umepeana kiunga cha \`thamani\` cha uga wa fomu bila kidhibiti cha \`onChange\`. Hii itatoa uga wa kusoma pekee.
 
 </ConsoleBlock>
 
-In the sandbox above, `ProductTable` and `SearchBar` read the `filterText` and `inStockOnly` props to render the table, the input, and the checkbox. For example, here is how `SearchBar` populates the input value:
+Katika kisanduku cha mchanga hapo juu, viunga `ProductTable` na `SearchBar` soma `filterText` na `inStockOnly` ili kutoa jedwali, ingizo, na kisanduku cha kuteua. Kwa mfano, hivi ndivyo `SearchBar` inavyojaza thamani ya ingizo:
 
 ```js {1,6}
 function SearchBar({ filterText, inStockOnly }) {
   return (
     <form>
       <input 
         type="text" 
-        value={filterText} 
+        thamani={filterText} 
```

Comment:
same as above, 
could we maintain the original language syntax i.e. in this case, the language syntax is in JavaScript